### PR TITLE
cloudflare_logpush_job: Add output_options support

### LINKED ISSUE #2173: Migrate logpush_job to use output_options
### Current Terraform and Cloudflare provider version

1.0.0

### Description

Cloudflare are replacing `logpull_options` with `output_options` which provides new functionality: https://developers.cloudflare.com/logs/reference/log-output-options/

Please can we support the new output_options format?

### Use cases

As a user, I would like to export log files as CSV files (which output_options allows) so that I can import them into my datawarehouse natively.

### Potential Terraform configuration

```resource "cloudflare_logpush_job" "name" {
  enabled             = true
  zone_id             = id
  destination_conf    = local.destination_conf
  dataset             = "http_requests"
  frequency           = "low"
  ownership_challenge = data.aws_s3_object.challenge_file_seo.body
  output_options = jsonencode({
    where = {
      field_names      = ["BotScore", "BotScoreSrc", "BotTags", "ClientRequestHost", "ClientRequestMethod", "ClientRequestURI", "ClientRequestPath", "ClientRequestUserAgent", "EdgeEndTimestamp", "EdgeResponseBytes", "EdgeResponseStatus", "EdgeStartTimestamp", "RayID", "ClientDeviceType", "ClientIPClass", "EdgeResponseContentType"]
      timestamp_format = "rfc3339"
      output_type      = "csv"
    }
  })
  filter = jsonencode({
    where = {
      key      = "ClientIPClass"
      operator = "eq"
      value    = "searchEngine"
    }
  })
}
```

### References

_No response_

### PR DESCRIPTION
**Pull request by dloebl**
Closes #2173.
[Cloudflare API documentation](https://developers.cloudflare.com/api/operations/post-accounts-account_identifier-logpush-jobs#request-body)

### CODE REVIEW

File: docs/resources/logpush_job.md [hunk 1]
```diff
@@ -125,13 +125,32 @@ resource "cloudflare_logpush_job" "example_job" {
...
 ### Read-Only
 
 - `id` (String) The ID of this resource.
 
+<a id="nestedblock--output_options"></a>
+### Nested Schema for `output_options`
+
+Optional:
+
+- `batch_prefix` (String)
```
Comment (jacobbednarz) [inline docs/resources/logpush_job.md:141]: can you please add `Description` values to the schema attributes to populate this correctly?

File: internal/sdkv2provider/schema_cloudflare_logpush_job.go [hunk 2]
```diff
@@ -128,5 +128,75 @@ func resourceCloudflareLogpushJobSchema() map[string]*schema.Schema {
...
+						Type:         schema.TypeFloat,
+						Optional:     true,
+						Default:      1.0,
+						ValidateFunc: validation.FloatBetween(0.0, 1.0),
+					},
+					"timestamp_format": {
+						Type:         schema.TypeString,
+						Optional:     true,
+						Default:      "unixnano",
+						ValidateFunc: validation.StringInSlice([]string{"unixnano", "unix", "rfc3339"}, false),
```
Comment (jacobbednarz) [inline internal/sdkv2provider/schema_cloudflare_logpush_job.go:196]: we should include this in the `Description` with the util helpers.

**Review by jacobbednarz — CHANGES_REQUESTED** [review]

File: docs/resources/logpush_job.md (hunk 1 above)
Comment (dloebl) [reply docs/resources/logpush_job.md]: Addressed with f27f6c3d563adbf7ffed82c304cc4a913f3dd165.

File: internal/sdkv2provider/schema_cloudflare_logpush_job.go (hunk 2 above)
Comment (dloebl) [reply internal/sdkv2provider/schema_cloudflare_logpush_job.go]: Addressed with f27f6c3d563adbf7ffed82c304cc4a913f3dd165.

**Comment by jacobbednarz** [comment]
thanks @dloebl 👏

**Pull request merged**